src/k8s-engine/lima.ts

```diff
@@ -1415,6 +1415,114 @@ export default class LimaBackend extends events.EventEmitter implements K8s.Kube
+    const contextContents = {
+      Name:      contextName,
+      Metadata:  { Description: 'Rancher Desktop moby context' },
+      Endpoints: {
+        docker: {
+          Host:          `unix://${ socketPath }`,
+          SkipTLSVerify: false,
+        },
+      } as Record<string, {Host: string, SkipTLSVerify: boolean, DefaultNamespace?: string}>,
+    };
+
+    if (kubernetesEndpoint) {
+      contextContents.Endpoints.kubernetes = {
+        Host:             kubernetesEndpoint,
+        SkipTLSVerify:    true,
+        DefaultNamespace: 'default',
+      };
+    }
+
+    console.debug(`Updating docker context: writing to ${ contextDir }`, contextContents);
+
+    await fs.promises.mkdir(contextDir, { recursive: true });
+    await fs.promises.writeFile(path.join(contextDir, 'meta.json'), JSON.stringify(contextContents));
+
+    try {
+      const existingConfig: {currentContext?: string} =
+        JSON.parse(await fs.promises.readFile(configPath, { encoding: 'utf-8' })) ?? {};
+
+      if (defaultSocket) {
+        // If we _are_ the default socket, we can just unset the current context
+        // (which will cause it to use the default)
+        if (existingConfig.currentContext) {
+          delete existingConfig.currentContext;
+          await fs.promises.writeFile(configPath, JSON.stringify(existingConfig));
+        }
+      } else if (existingConfig.currentContext !== contextName) {
+        // We don't have the default socket, and the existing config doesn't
+        // exist or isn't pointing at our context.
+        // We should look up the current context, and check if it's valid; if
+        // (and only if) it's not valid, then set the default context to ours.
+        let existingSocket = `unix://${ DEFAULT_DOCKER_SOCK_LOCATION }`;
+
+        if (existingConfig.currentContext) {
+          for (const dir of await fs.promises.readdir(contextParent)) {
+            const dirPath = path.join(contextParent, dir, 'meta.json');
+
+            try {
+              const data = yaml.parse(await fs.promises.readFile(dirPath, 'utf-8'));
+
+              if (data.Name === existingConfig.currentContext) {
+                existingSocket = data.Endpoints?.docker?.Host as string ?? '';
+                break;
+              }
+            } catch (ex) {
+              console.log(`Failed to read context ${ dir }, skipping: ${ ex }`);
+            }
+          }
+          if (!existingSocket.startsWith('unix://')) {
+            // Using a non-unix socket (e.g. TCP); assume it's working fine.
+            return;
+          }
+          existingSocket = existingSocket.replace(/^unix:\/\//, '');
+          try {
+            const stat = await fs.promises.stat(existingSocket);
+
+            if (stat.isSocket()) {
+              return;
+            }
+            console.log(`Invalid existing context "${ existingConfig.currentContext }": ${ existingSocket } is not a socket; overriding.`);
```

Comment:
I don't understand the log messages here and at line 1505 -- where is `existingSocket` overridden?

Reply:
`existingSocket` defaults to `unix:///var/run/docker.sock`; if any of the stored docker contexts match the current context, it's overridden to be the socket given in the context instead (line 1485).
We then strip off the `unix://` prefix in line 1496 (but abort if it's not a unix socket at all, line 1492).
We can then check if that socket is valid (exists and is a unix socket); if not, we override (because using a broken socket isn't useful).

Reply:
I think the mixup is in what's being overridden -- it isn't `existingSocket` that gets overwritten, it seems to just be `existingConfig.currentContext` set to `contextName`. Could you update the comment to be more precise on what's being overridden